pymc3/tests/test_theanof.py

```diff
@@ -74,3 +77,23 @@ def gen():
         np.testing.assert_equal(np.ones((10, 10)) * 0, f())
         np.testing.assert_equal(np.ones((10, 10)) * 1, f())
         self.assertRaises(StopIteration, f)
+
+    def test_pickling(self):
+        data = np.random.uniform(size=(1000, 10))
+        minibatches = DataSampler(data, batchsize=50)
+        gen = generator(minibatches)
+        pickle.loads(pickle.dumps(gen))
+        bad_gen = generator(integers())
+        self.assertRaises(Exception, pickle.dumps, bad_gen)
+
+    def test_gen_cloning_with_shape_change(self):
+        data = np.random.uniform(size=(1000, 10))
```

Comment:
@springcoil this test appeared from a problem that I'm still not able to reproduce. 
After some manipulations with `theano.clone` I have not replaced generator. Thus I was not able to set a new input with new shape (e.g. shared variable)